django/contrib/admin/views/main.py

```diff
@@ -57,16 +69,19 @@ def __init__(self, request, model, list_display, list_display_links,
         self.sortable_by = sortable_by
 
         # Get search parameters from the query string.
-        try:
-            self.page_num = int(request.GET.get(PAGE_VAR, 0))
-        except ValueError:
-            self.page_num = 0
+        change_list_form = ChangeListForm(request.GET)
+        if not change_list_form.is_valid():
+            messages.error(request, change_list_form.errors.as_text())
+
+        self.page_num = change_list_form.cleaned_data.get(PAGE_VAR) or 0
```

Comment:
[`get()` takes a default to return as a second parameter](https://docs.python.org/3.7/library/stdtypes.html#dict.get).

Reply:
this is about page_num param. I couldn't find but the page_num value may be None. So I did like this.
```
>>> x = {'a': 1, 'b': None}
>>>x.get('b', 1)
None
>>>x.get('b', 1) or 1
1
>>>x.get('c', 1)
1
>>> 
```